drivers/src/data_vault.rs

```diff
@@ -390,14 +391,19 @@ impl DataVault {
         self.read_cold_reset_entry4(ColdResetEntry4::LmsVendorPubKeyIndex)
     }
 
-    /// Set and lock the rom cold boot status.
+    /// Set the rom cold boot status.
     ///
     /// # Arguments
     ///
     /// * `status` - Rom Cold Boot Status
+    /// * `lock_entry` - Lock the entry
     ///
-    pub fn set_rom_cold_boot_status(&mut self, status: u32) {
-        self.write_lock_cold_reset_entry4(ColdResetEntry4::RomColdBootStatus, status);
+    pub fn set_rom_cold_boot_status(&mut self, status: u32, lock_entry: bool) {
```

Comment:
It's not clear to the reader of calling code what these booleans mean. Perhaps this should follow the same pattern as write_lock_cold_reset_entry4() and be two separate functions?

`set_rom_cold_boot_status()` and `set_lock_cold_boot_status()`?

Reply:
Although I'm honestly not sure why we're not just calling the write_*entry4 functions directly....

Reply:
Removed the additional helper methods.